internal/api/grpc/system/instance_converter.go

```diff
@@ -3,17 +3,109 @@ package system
 	"github.com/zitadel/zitadel/internal/api/grpc/object"
+	z_oidc "github.com/zitadel/zitadel/internal/api/oidc"
 	"github.com/zitadel/zitadel/internal/command"
 	"github.com/zitadel/zitadel/internal/domain"
 	"github.com/zitadel/zitadel/internal/query"
 	instance_pb "github.com/zitadel/zitadel/pkg/grpc/instance"
 	system_pb "github.com/zitadel/zitadel/pkg/grpc/system"
 )
 
+func CreateInstancePbToSetupInstance(req *system_pb.CreateInstanceRequest, defaultInstance command.InstanceSetup, externalDomain string) *command.InstanceSetup {
+	if req.InstanceName != "" {
+		defaultInstance.InstanceName = req.InstanceName
+		defaultInstance.Org.Name = req.InstanceName
+	}
+	if req.CustomDomain != "" {
+		defaultInstance.CustomDomain = req.CustomDomain
+	}
+	if req.FirstOrgName != "" {
+		defaultInstance.Org.Name = req.FirstOrgName
+	}
+
+	if user := req.GetMachine(); user != nil {
+		defaultInstance.Org.Machine = createInstancePbToAddMachine(user)
+		defaultInstance.Org.Human = nil
+	}
+	if user := req.GetHuman(); user != nil {
+		defaultInstance.Org.Human = createInstancePbToAddHuman(user, defaultInstance.DomainPolicy.UserLoginMustBeDomain, defaultInstance.Org.Name, externalDomain)
+		defaultInstance.Org.Machine = nil
+	}
+
+	if lang := language.Make(req.DefaultLanguage); lang != language.Und {
+		defaultInstance.DefaultLanguage = lang
+	}
+
+	return &defaultInstance
+}
+func createInstancePbToAddHuman(user *system_pb.CreateInstanceRequest_Human, userLoginMustBeDomain bool, org, externalDomain string) *command.AddHuman {
+	human := &command.AddHuman{}
+	if user.Email != nil {
+		human.Email.Address = user.Email.Email
+		human.Email.Verified = user.Email.IsEmailVerified
+	}
+	if user.Profile != nil {
+		if user.Profile.FirstName != "" {
+			human.FirstName = user.Profile.FirstName
+		}
+		if user.Profile.LastName != "" {
+			human.LastName = user.Profile.LastName
+		}
+		if user.Profile.PreferredLanguage != "" {
+			lang, err := language.Parse(user.Profile.PreferredLanguage)
+			if err == nil {
+				human.PreferredLanguage = lang
+			}
+		}
+	}
+	// check if default username is email style or else append @<orgname>.<custom-domain>
+	// this way we have the same value as before changing `UserLoginMustBeDomain` to false
+	if !userLoginMustBeDomain && !strings.Contains(human.Username, "@") {
+		human.Username = human.Username + "@" + domain.NewIAMDomainName(org, externalDomain)
+	}
+	if user.UserName != "" {
+		human.Username = user.UserName
+	}
+	if user.Password != nil {
+		human.Password = user.Password.Password
+		human.PasswordChangeRequired = user.Password.PasswordChangeRequired
+	}
+	return human
+}
+
+func createInstancePbToAddMachine(user *system_pb.CreateInstanceRequest_Machine) *command.AddMachine {
+	machine := &command.AddMachine{
+		Machine: &command.Machine{},
+	}
+	if user.UserName != "" {
+		machine.Machine.Username = user.UserName
+	}
+	if user.Name != "" {
+		machine.Machine.Name = user.Name
+	}
```

Comment:
these ifs are unnecessary

Reply:
all unnecessary ifs are removed

Reply:
with the refactored code, the ifs would now make sense 😄